packages/api-client/src/client/types.ts

```diff
@@ -66,5 +66,36 @@ export interface ApiFacets {
 
 export interface ApiQuery<T> {
     total_count?: number;
-    results: T[];
+    records: T[];
+}
+
+export enum ExportCatalogFormat {
```

Comment:
I'm more and more using: 
```
ExportCatalogFormat = {
    ...
} as const;
ExportCalalogFormat type = typeof ExportCatalogFormat[keyof typeof ExportCatalogFormat];
```
It seems a bit verbose, but objects are useable in vanilla JS (or enums are exported as objects maybe?) and more often than not, you end up building the object and according type anyways.

Reply:
(got that from the [TS doc](https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums) by the way)

Reply:
I got an error when building the package and I can't see why it would say that considering the code:
```
@opendatasoft/api-client: (!) Plugin typescript: @rollup/plugin-typescript TS2345: Argument of type '"geojson"' is not assignable to parameter of type 'ExportDatasetFormat'.
```
I get issues with TS and enums so many times, I'm starting to think enums are more trouble than they're worth in some cases.

If we got too many issues we could also use a good one `ExportCatalogFormat = 'json' | ' csv';` etc etc.

But maybe Etienne's suggestion fixes this already

Reply:
I didn't know this const TS trick.  I'm trying it